info.go

```diff
@@ -61,6 +62,10 @@ func newMapInfoFromProc(fd *internal.FD) (*MapInfo, error) {
 // The pointer fields are not supported across all kernels, and may be nil.
 type ProgramInfo struct {
 	Type ProgramType
+	// Truncated hash of the BPF bytecode.
+	Tag *string
+	// Name as supplied by user space at load time.
+	Name *string
```

Comment:
Can this be just a `string` that is empty when attribute is not available?

Reply:
Yeah, then it's not possible to distinguish unnamed vs. no names supported. But maybe that distinction is useless in the first place.

We could do the same thing for tag, and define `""` to mean "tag is not supported" but then the naive comparison `a.Tag == b.Tag` does the wrong thing (maybe?) on older kernels where Tag is not available.

Reply:
The user has to be mindful either way. Whether its to check for `nil` pointer or for empty `string`.